api/client/service.proto

```diff
@@ -34,10 +36,62 @@ message ExecutionResponse {
   google.rpc.Status error_detail = 7;
 }
 
+// Used to request a RequestStreamResponse.
+message RequestStreamRequest {
+  // Indicates the amount of RequestStreamResponses that should be yielded by the server
+  // in response to this request.
+  // Used to implement basic flow control by the client. Once AsyncClientImpl gets a way
+  // to apply backpressure this could probably be dropped.
+  uint64 amount = 1;
+}
+
+message RequestStreamResponse {
+  // Specifies what the requests look like. Will be merged with the client's configuration.
+  RequestSpecifier request_specifier = 1;
+  // Response-level expectations associated to the above request specification.
+  Expectations expectations = 2;
+}
+
+message RequestSpecifier {
+  // method/path/authority/content_length can all be expressed in request headers.
+  // so technically they are redundant. In case of ambiguity, the explicit fields will prevail.
+  // When set, these fields will replace the defaults configured on the remote side.
+  // Note that as of today, the target upstream/cluster that will be used to replay traffic to
+  // fixes the destination address/port.
+
+  // Request method
+  google.protobuf.StringValue method = 1;
+  // Request path
+  google.protobuf.StringValue path = 2;
+  // Request authority
+  google.protobuf.StringValue authority = 3;
+  // Request content length. The client will transfer the number of bytes specified here for the
+  // request body.
+  google.protobuf.UInt32Value content_length = 4;
```

Comment:
Why is this wrapped? Doesn't i have a default value of 0?

Reply:
The other side overrides its configured values with inbound values originating from this. 
While doing so it needs to be able to discern between not-set and `0`. 
Concretely, if we omit `content_length` here, the idea is that the other side will use its configured value and
not have it overridden to `0`.

Reply:
I believe this was answered, I do see a value in giving the server the ability to see which of these fields were set. Planning to proceed and merge this in. @htuch, please let me know if I misread this and you would like us to make further changes to this API. We can still iterate on it in #137.